File: split.js
const fs = require('fs');

try {
  let html = fs.readFileSync('portfolio-3d.html', 'utf8');

  // Extract CSS
  const styleRegex = /<style>([\s\S]*?)<\/style>/;
  const styleMatch = html.match(styleRegex);
  if(styleMatch) {
      fs.writeFileSync('portfolio-3d.css', styleMatch[1].trim() + '\n');
      html = html.replace(styleRegex, '<link rel="stylesheet" href="portfolio-3d.css"/>');
  }

  // Extract JS (the last script tag which contains our code)
  const scriptRegex = /<script>\s*?\n([\s\S]*?)\n<\/script>/;
  const scriptMatch = html.match(scriptRegex);
  if(scriptMatch) {
      fs.writeFileSync('portfolio-3d.js', scriptMatch[1].trim() + '\n');
      html = html.replace(scriptRegex, '<script src="portfolio-3d.js"></script>');
  } else {
      console.log("Could not find the script block using regular expression.");
  }

  fs.writeFileSync('portfolio-3d.html', html);
  console.log("Successfully split the files.");
} catch(e) {
  console.error(e);
}

File: split.py
import re

try:
    with open('portfolio-3d.html', 'r', encoding='utf-8') as f:
        html = f.read()

    # Extract CSS
    style_match = re.search(r'<style>([\s\S]*?)</style>', html)
    if style_match:
        css_content = style_match.group(1).strip() + '\n'
        with open('portfolio-3d.css', 'w', encoding='utf-8') as f:
            f.write(css_content)
        html = re.sub(r'<style>([\s\S]*?)</style>', '<link rel="stylesheet" href="portfolio-3d.css"/>', html)

    # Extract JS
    script_match = re.search(r'<script>\s*\n([\s\S]*?)</script>', html)
    if script_match:
        js_content = script_match.group(1).strip() + '\n'
        with open('portfolio-3d.js', 'w', encoding='utf-8') as f:
            f.write(js_content)
        html = re.sub(r'<script>\s*\n([\s\S]*?)</script>', '<script src="portfolio-3d.js"></script>', html)
    else:
        print("Could not find the script block.")

    with open('portfolio-3d.html', 'w', encoding='utf-8') as f:
        f.write(html)
        
    print("Successfully separated the files using Python.")
except Exception as e:
    print(f"Error: {e}")
